### PR TITLE
gRPC2 migration

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Switch to gRPC 2.

## Changes

- Migrate to using gRPC with the web sdk.
- Cleanup gRPC proto building.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Comment by orhoj** [comment]
I am unable to connect to a node on this branch.

**Comment by orhoj** [comment]
I tried to bump `node` version for the build pipeline to new minimum requirement (from one of our dependencies), but it appears to give some other issues while building. So we need to find a version that will work with everything.

File: app/utils/transactionHelpers.ts [hunk 1]
```diff
@@ -571,8 +572,12 @@ export function buildTransactionAccountSignature(
     return transactionAccountSignature;
 }
 
-export function isSuccessfulTransaction(event: TransactionSummary) {
-    return event.result.outcome === 'success';
+export function isSuccessfulTransaction(summary: BlockItemSummary) {
+    if (summary.type === TransactionSummaryType.AccountTransaction) {
+        return summary.transactionType !== TransactionKindString.Failed;
+    }
+    // other block items don't fail // TODO is this completely true
```
Comment (orhoj) [inline app/utils/transactionHelpers.ts:579]: We need to resolve this TODO.

File: app/utils/withUpdateQueue.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                <Execute
+                    execution={init}
+                    onError={() =>
+                        dispatch(
+                            push({ pathname: routes.MULTISIGTRANSACTIONS })
+                        )
+                    }
+                    onSuccess={setChainData}
+                    errorTitle="Error communicating with node"
+                    errorContent="We were unable to retrieve the block summary from the
```
Comment (orhoj) [inline app/utils/withUpdateQueue.tsx:46]: Error message mentions block summary, but that is not correct.

File: jest.config.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ module.exports = {
             '<rootDir>/internals/mocks/fileMock.js',
         '\\.(css|less|sass|scss)$': 'identity-obj-proxy',
         '^~/(.*)$': '<rootDir>/app/$1',
+        '@concordium/web-sdk': '@concordium/common-sdk',
```
Comment (orhoj) [inline jest.config.js:10]: Why is this needed?

File: package.json [hunk 4]
```diff
@@ -251,8 +248,7 @@
     "yargs": "^17.1.1"
   },
   "dependencies": {
-    "@concordium/node-sdk": "^1.0.0-alpha.0",
-    "@grpc/grpc-js": "^1.3.6",
+     "@concordium/web-sdk": "^6.4.0",
```
Comment (orhoj) [inline package.json:251]: Broken formatting.

**Review by orhoj — CHANGES_REQUESTED** [review]

File: app/node/nodeHelpers.ts [hunk 5]
```diff
@@ -92,17 +80,12 @@ export async function fetchGlobal(specificBlockHash?: string): Promise<Global> {
...
 }
 
 export async function getEnergyToMicroGtuRate(): Promise<Fraction> {
-    const consensusStatus = await getConsensusStatus();
-    const blockSummary = await getBlockSummary(
-        consensusStatus.lastFinalizedBlock
-    );
-    const { euroPerEnergy } = blockSummary.updates.chainParameters;
-    const { microGTUPerEuro } = blockSummary.updates.chainParameters;
+    const params = await getBlockChainParameters(undefined);
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/node/nodeHelpers.ts:87]: I guess not

File: app/pages/multisig/common/RelativeRateField/util.ts [hunk 6]
```diff
@@ -3,8 +3,9 @@ import { Validate } from 'react-hook-form';
 import { isValidBigInt } from '~/utils/numberStringHelpers';
 import { ExchangeRate } from '~/utils/types';
 
+// TODO why was this type string / undefined?
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/pages/multisig/common/RelativeRateField/util.ts:6]: I don't remember, I guess I'll revert this.

File: app/pages/multisig/updates/ElectionDifficulty/ElectionDifficultyView.tsx [hunk 7]
```diff
@@ -14,14 +19,23 @@ interface Props extends ChainData {
...
 
-    const currentElectionDifficulty =
-        blockSummary.updates.chainParameters.electionDifficulty;
+    if (
+        (!isChainParametersV0(chainParameters) &&
+            !isChainParametersV1(chainParameters)) ||
+        !isConsensusStatusV0(consensusStatus)
+    ) {
+        throw new Error(
+            'election difficulty only available on Protocol versions < 6'
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/pages/multisig/updates/ElectionDifficulty/ElectionDifficultyView.tsx:34]: Well we usually keep the updates around :thinking:

File: jest.config.js (hunk 3 above)
Comment (shjortConcordium) [reply jest.config.js:10]: I don't remember, maybe because of the wasm module, but tests just don't work directly on web-sdk, I had to do the same for BW.

File: app/utils/transactionHelpers.ts (hunk 1 above)
Comment (shjortConcordium) [reply app/utils/transactionHelpers.ts:579]: Yes, it is true.
The only reason to have failed account transactions is to make the account pay, but that is not relevant for other types, so they wouldn't be put in chain.

File: app/utils/transactionHash.ts [hunk 8]
```diff
@@ -25,12 +25,11 @@ export async function getUpdateInstructionTransactionHash(
     updateInstruction: UpdateInstruction
 ) {
     const handler = findUpdateInstructionHandler(updateInstruction.type);
-    const blockSummary = (await fetchLastFinalizedBlockSummary())
-        .lastFinalizedBlockSummary;
+    const keys = await getBlockChainParameters();
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/utils/transactionHash.ts:28]: Sure, I've just changed it to `chainParameters`.

File: app/pages/multisig/updates/MintDistribution/util.ts [hunk 9]
```diff
@@ -20,7 +21,11 @@ export const toRewardDistributionValue = ({
...
 export const getSlotsPerYear = (consensusStatus: ConsensusStatus): number => {
-    const slotsPerSecond = 1000 / Number(consensusStatus.slotDuration);
-    const slotsPerYear = slotsPerSecond * 60 * 60 * 24 * 365.25;
-    return slotsPerYear;
+    if (isConsensusStatusV0(consensusStatus)) {
+        const slotsPerSecond = 1000 / Number(consensusStatus.slotDuration);
+        const slotsPerYear = slotsPerSecond * 60 * 60 * 24 * 365.25;
+        return slotsPerYear;
+    }
+    // TODO how to calculate this?
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/pages/multisig/updates/MintDistribution/util.ts:29]: I think I have resolved it.

File: app/pages/multisig/updates/GasRewards/GasRewardsForm.tsx [hunk 10]
```diff
@@ -65,6 +65,7 @@ export default function GasRewardsForm({
     const firstError = fields.map((f) => errors[f]).filter(isDefined)[0]
         ?.message;
 
+    // TODO handle when finalizationProof is not there, correctly
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/pages/multisig/updates/GasRewards/GasRewardsForm.tsx:68]: During the task where we update the existing updates. I'll remove the TODO so it doesn't end up hanging.

File: app/constants/transactionKindNames.json [hunk 11]
```diff
@@ -23,5 +23,6 @@
     "registerData": "Data registration",
     "configureBaker": "Configure baker",
     "configureDelegation": "Configure delegation",
-    "paydayAccountReward": "Reward payout"
+    "paydayAccountReward": "Reward payout",
+    "failed": "Failed Transaction"
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/constants/transactionKindNames.json:27]: Yes, it always has been, we just haven't handled it, I guess :thinking:

File: app/node/nodeHelpers.ts [hunk 12]
```diff
@@ -112,17 +95,19 @@ export async function getEnergyToMicroGtuRate(): Promise<Fraction> {
     return { numerator, denominator };
 }
 
+// TODO Is there a better way to do this in gRPC 2
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/node/nodeHelpers.ts:98]: Yes, I've added to the SDK, it the health check. :thinking:

File: app/features/TransactionSlice.ts [hunk 13]
```diff
@@ -119,6 +121,8 @@ function shieldedOnlyFilter(filter: TransactionFilter): TransactionFilter {
         configureBaker: false,
         configureDelegation: false,
         paydayAccountReward: false,
+        // TODO should this not be added?
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/features/TransactionSlice.ts:124]: I believe the failed is only when it is not possible to determine the transaction type, so it shouldn't be shown in shielded.

File: app/features/ChainDataSlice.ts [hunk 14]
```diff
@@ -1,23 +1,33 @@
...
 import { createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Dispatch } from 'redux';
 import { getConsensusStatus } from '~/node/nodeRequests';
 import type { RootState } from '~/store/store';
 import { pipe } from '~/utils/basicHelpers';
 import { orUndefined } from '~/utils/functionHelpers';
 
-type CSDateKey = KeysMatching<ConsensusStatus, Date>;
-type CSBigIntKey = KeysMatching<ConsensusStatus, bigint>;
+// TODO test this with v0 and v1
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/features/ChainDataSlice.ts:13]: Hmm?
Removed the TODOS, It was a weird place to put this TODO about testing the wallet with old and new consensus :thinking: 

I have not used the BFTStatus, so lets not save it.

File: app/features/TransactionSlice.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (orhoj) [reply app/features/TransactionSlice.ts:124]: Okay, so we can remove the TODO?

[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/features/TransactionSlice.ts:124]: Yeah, I will remove it.

File: app/node/nodeHelpers.ts (hunk 12 above)
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/node/nodeHelpers.ts:98]: I've not added this, the node gets angry when I use health check endpoint :thinking:

File: app/pages/multisig/updates/MintDistribution/util.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply app/pages/multisig/updates/MintDistribution/util.ts]: (It only needs to be calculated for v0, because afterwards the related field is removed from the update)

File: app/shell/Status/NodeStatus.tsx [hunk 15]
```diff
@@ -57,10 +57,16 @@ export default function NodeStatus(): JSX.Element {
...
-                const upToDate = await isNodeUpToDate();
-                status = upToDate
-                    ? NodeConnectionStatus.Ready
-                    : NodeConnectionStatus.CatchingUp;
+                const healthResponse = await healthCheck();
+                if (healthResponse.isHealthy) {
+                    status = NodeConnectionStatus.Ready;
+                } else if (
+                    healthResponse.message?.includes(
+                        'Last finalized block is too far behind'
```
Comment (orhoj) [inline app/shell/Status/NodeStatus.tsx:65]: Is there no better way to check this? This seems a little fragile against changes to the text. Are there any other types of health responses that would be relevant?

Comment (shjortConcordium) [reply app/shell/Status/NodeStatus.tsx:65]: There is no other way to check this, the same error code is when we can't connect at all. :cry:

**Review by orhoj — APPROVED** [review]
